src/Writer/SnowflakeWriteAdapter.php

```diff
@@ -5,60 +5,107 @@
+use Keboola\Temp\Temp;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+use SplFileInfo;
+use Symfony\Component\Process\Process;
 
 /**
  * @property-read SnowflakeQueryBuilder $queryBuilder
  */
 class SnowflakeWriteAdapter extends OdbcWriteAdapter
 {
-    public function writeData(string $tableName, ExportConfig $exportConfig): void
+    use QuoteTrait;
+
+    private Temp $tempDir;
+
+    private SplFileInfo $snowSqlConfig;
+
+    public function __construct(
+        SnowflakeConnection $connection,
+        QueryBuilder $queryBuilder,
+        LoggerInterface $logger,
+    ) {
+        parent::__construct($connection, $queryBuilder, $logger);
+
+        $this->tempDir = new Temp('wr-snowflake-adapter');
+    }
+
+    public function getName(): string
     {
-        $this->logger->info(sprintf('Writing data to table "%s"', $tableName));
+        return 'Snowsql';
+    }
 
+    public function writeData(string $tableName, ExportConfig $exportConfig): void
+    {
         /** @var SnowflakeDatabaseConfig $databaseConfig */
         $databaseConfig = $exportConfig->getDatabaseConfig();
 
-        $stageName = $this->generateStageName($databaseConfig->hasRunId() ? $databaseConfig->getRunId() : '');
+        $this->snowSqlConfig = $this->createSnowSqlConfig($databaseConfig);
 
-        $this->logger->info(sprintf('Dropping stage "%s"', $stageName));
-        $this->connection->exec($this->queryBuilder->dropStageStatement($this->connection, $stageName));
+        // Upload to internal stage
+        $this->logger->info(sprintf('Uploading data to internal stage "@~/%s"', $tableName));
+        $this->putIntoInternalStage($exportConfig, $tableName);
 
-        $writeStrategy = $this->getTableWriteStrategy($exportConfig->getTableFilePath());
-
-        $this->logger->info(sprintf('Creating stage "%s"', $stageName));
-        $this->connection->exec($writeStrategy->generateCreateStageCommand($stageName));
-
-        $tableNameWithSchema = sprintf(
-            '%s.%s',
-            $this->connection->quoteIdentifier($databaseConfig->getSchema()),
-            $this->connection->quoteIdentifier($tableName),
-        );
         try {
             $items = array_filter(
                 $exportConfig->getItems(),
                 fn(ItemConfig $item) => strtolower($item->getType()) !== 'ignore',
             );
-            $commands = $writeStrategy->generateCopyCommands(
-                tableName: $tableNameWithSchema,
-                stageName: $stageName,
-                items: $items,
-            );
-            foreach ($commands as $command) {
-                $this->connection->exec($command);
-            }
+
+            // Copy from internal stage to staging table
+            $this->logger->info(sprintf('Copying data from internal stage to staging table "%s"', $tableName));
+            $query = $this->queryBuilder->copyIntoTableQueryStatement($this->connection, $tableName, $items);
+            $this->logger->debug($query);
```

Comment:
to co běží přes ODBC Driver tak nelobuj (bloguje se automaticky tady - https://github.com/keboola/db-writer-adapter/blob/main/src/Connection/BaseConnection.php#L79)

Reply:
jo, já už si tu dementní automatickou opravu textu musím vypnout :D